s3_benchmark.py

```diff
@@ -0,0 +1,179 @@
+        # Read into memoryview m.
+        self._available_stream()
+        len = self._stream.readinto1(m)
+        self._statistic.record_read(len)
+        if len is 0:
+            self._stream.close()
+        return len
+
+    def seek(self, offset, whence):
+        self._available_stream()
+        return self._stream.seek(offset, whence)
+
+    def close(self):
+        pass
+
+
+class Statistics(object):
+
+    def __init__(self):
+        self._lock = threading.Lock()
+        self.end_time = 0
+        self._bytes_peak = 0
+        self._bytes_avg = 0
+        self._bytes_read = 0
+        self._bytes_sampled = 0
+        self.sec_first_byte = 0
+        self.star_time = time.time()
+        self.last_sample_time = time.time()
+
+    def record_read(self, size):
+        with self._lock:
+            self._bytes_read += size
+            if self.sec_first_byte == 0:
+                self.sec_first_byte = time.time() - self.star_time
+            time_now = time.time()
+            if time_now - self.last_sample_time > 1:
+                bytes_this_second = (self._bytes_read - self._bytes_sampled) / (time_now - self.last_sample_time)
+                self._bytes_sampled = self._bytes_read
+                self._bytes_avg = (self._bytes_avg + bytes_this_second) * 0.5
+                if self._bytes_peak < bytes_this_second:
+                    self._bytes_peak = bytes_this_second
+                self.last_sample_time = time_now
+
+    def bytes_peak(self):
+        return (self._bytes_peak * 8) / GBPS
+
+    def bytes_avg(self):
+        return (self._bytes_avg * 8) / GBPS
+
+
+# Configurations
+region = "us-west-2"
+bucket_name = "aws-crt-python-s3-testing-bucket"
+object_name = "/0_10GB.txt"
+file_name = "." + object_name
+object_real_name = "/0_10GB"
+suffix = ".txt"
+repeat_times = 1
+bunch_size = 160
+
+writing_disk = True
+request_type = "download"
+
+# Initialization
+event_loop_group = EventLoopGroup(18)
+host_resolver = DefaultHostResolver(event_loop_group)
+bootstrap = ClientBootstrap(event_loop_group, host_resolver)
+credential_provider = AwsCredentialsProvider.new_default_chain(bootstrap)
+s3_client = S3Client(
+    bootstrap=bootstrap,
+    region="us-west-2",
+    credential_provider=credential_provider,
+    throughput_target_gbps=100)
+
+t_statistic = Statistics()
+
+headers = HttpHeaders([("host", bucket_name + ".s3." + region + ".amazonaws.com")])
+request = HttpRequest("GET", object_name, headers)
+
+# file_stats = os.stat(file_name)
+# data_len = file_stats.st_size
+
+# data_stream = CrtLazyReadStream(file_name, "r+b", t_statistic, data_len)
+# upload_headers = HttpHeaders([("host", bucket_name + ".s3." + region + ".amazonaws.com"),
+#                               ("Content-Type", "text/plain"), ("Content-Length", str(data_len))])
+# upload_request = HttpRequest("PUT", "/put_object_test_py_10MB.txt", upload_headers, data_stream)
+
+
+def on_body(offset, chunk, **kwargs):
+    t_statistic.record_read(len(chunk))
```

Comment:
isn't chunk NULL when using the FILE* api?

Reply:
The on_body callback will not be invoked when the FILE pointer is set.